src/App.tsx

```diff
@@ -1,29 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemList } from './components/ItemsList';
+import { Option } from './components/Option';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+export const items = getNumbers(1, 42).map(n => `Item ${n}`);
+export const options = [3, 5, 10, 20, 40];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handleChangePages = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {`${currentPage}`} (items{' '}
+        {`${currentPage * perPage - perPage + 1} - ${currentPage * perPage > items.length ? items.length : currentPage * perPage}`}{' '}
+        of 42)
```

Comment:
The hard-coded number `42` should ideally be replaced with a dynamic value such as `items.length` to make the code more maintainable and to avoid potential errors if the number of items changes.